### PR TITLE
feat(types): infer options in plugins

### LINKED ISSUE #269: Expose typings for Plugins
I think it would be helpful to expose the `Instance` type so plugins can type the argument (https://vue-test-utils.vuejs.org/v2/guide/plugins.html#writing-a-plugin) as well as adapting the `install` function to infer the options parameter:

```ts
    install(handler: (instance: Instance) => Record<string, any>, options?: Record<string, any>): void;
```

Similar to https://github.com/vuejs/vue-next/pull/2758

### PR DESCRIPTION
**Pull request by posva**
Close #269

### CODE REVIEW

File: test-dts/plugins.d-test.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+config.plugins.VueWrapper.install(PluginWithOptions, { msg: 'Hello' })
+config.plugins.VueWrapper.install(PluginWithoutOptions)
+config.plugins.VueWrapper.install(PluginWithOptionalOptions)
+config.plugins.VueWrapper.install(PluginWithOptionalOptions2)
+config.plugins.VueWrapper.install(PluginWithOptionalOptions, { msg: 'hello' })
+config.plugins.VueWrapper.install(PluginWithOptionalOptions2, { msg: 'hello' })
+
+// FIXME: not sure if this one is possible
+// expectError(config.plugins.VueWrapper.install(PluginWithoutOptions, {}))
```
Comment (cexbrayat) [inline test-dts/plugins.d-test.ts:42]: I guess this is fine?
The FIXME needs to be removed before merging.

**Review by cexbrayat — APPROVED** [review]

File: test-dts/plugins.d-test.ts (hunk 1 above)
Comment (posva) [reply test-dts/plugins.d-test.ts:42]: I think it's fine too but I thought I might have overlooked a way of doing it in TS. If there is nothing to fix, yeah, better remove it 😆

Comment (cexbrayat) [reply test-dts/plugins.d-test.ts:42]: We _could_ probably do something, but I'm not sure it's worth the time, the hassle and the maintenance cost 😉 Thanks for the PR!

File: test-dts/plugins.d-test.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+config.plugins.VueWrapper.install(PluginWithOptions, { msg: 'Hello' })
+config.plugins.VueWrapper.install(PluginWithoutOptions)
+config.plugins.VueWrapper.install(PluginWithOptionalOptions)
+config.plugins.VueWrapper.install(PluginWithOptionalOptions2)
+config.plugins.VueWrapper.install(PluginWithOptionalOptions, { msg: 'hello' })
+config.plugins.VueWrapper.install(PluginWithOptionalOptions2, { msg: 'hello' })
+
+// FIXME: not sure if this one is possible
```
Comment (posva) [inline test-dts/plugins.d-test.ts:41]:
```suggestion
// uncertain if it is possible to forbid this usage
```

**Review by lmiller1990 — APPROVED** [review]
great, thanks!

**Pull request merged**